### PR TITLE
refactor(editor): improve naming and comments

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
I'm not sure if this is the best explanation, @ShaunSHamilton @nhcarrigan, but I think it's a slight improvement.  Any ideas are welcome!

Ref: https://github.com/freeCodeCamp/freeCodeCamp/pull/43617#discussion_r719370771

### CODE REVIEW

File: client/src/templates/Challenges/classic/editor.tsx [hunk 1]
```diff
@@ -646,15 +646,19 @@ const Editor = (props: EditorProps): JSX.Element => {
...
       initializeRegions(getEditableRegionFromRedux());
       addWidgetsToRegions(editor);
     }
   }
 
-  function isProject() {
+  // The challenge object always has the editableRegionBoundaries array, but
+  // unless it is a step in the project based curriculum it will be an empty
+  // array. That's currently how we determine if we need the extra features
+  // (multiple editor tabs, embedded description, etc.) or not.
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/classic/editor.tsx:660]:
```suggestion
  // Currently, only practice project parts have editable region markers
  // This functions is used to enable multiple editor tabs, jaws, etc.
```

File: client/src/templates/Challenges/classic/editor.tsx [hunk 2]
```diff
@@ -646,15 +646,17 @@ const Editor = (props: EditorProps): JSX.Element => {
...
     const editor = data.editor;
     if (editor) {
       initializeRegions(getEditableRegionFromRedux());
       addWidgetsToRegions(editor);
     }
   }
 
-  function isProject() {
+  // Currently, only practice project parts have editable region markers
+  // This functions is used to enable multiple editor tabs, jaws, etc.
```
Comment (Sembauke) [inline client/src/templates/Challenges/classic/editor.tsx:658]:
```suggestion
  // These functions is used to enable multiple editor tabs, jaws, etc.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: client/src/templates/Challenges/classic/editor.tsx (hunk 2 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/classic/editor.tsx:658]: "This function" refers to `isProject`, so there's only one.

Comment (Sembauke) [reply client/src/templates/Challenges/classic/editor.tsx:658]: got it.

Comment (Sembauke) [reply client/src/templates/Challenges/classic/editor.tsx:658]: I updated my requested change accordingly

**Review by Sembauke — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**